flag/value.go

```diff
@@ -0,0 +1,82 @@
+package flag
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/urso/ucfg"
+)
+
+// NewFlagKeyValue implements the flag.Value interface for
+// capturing ucfg.Config settings from command line arguments.
+// Configuration options follow the argument name and must be in the form of
+// "key=value". Using 'D' as command line flag for example, options on command line
+// must be given as:
+//
+// -D key1=value -D key=value
+//
+// Note: the space between command line option and key is required by the flag
+// package to parse command line flags correctly.
+//
+// Note: it's valid to use a key multiple times. If keys are used multiple
+// times, values get overwritten. The last known value for some key will be stored
+// in the generated configuration.
+//
+// The type of value must be any of bool, uint, int, float, or string. Any kind
+// of array or object syntax is not supported.
+//
+// If autoBool is enabled (default if Config or ConfigVar is used), keys without
+// value are converted to bool variable with value being true.
```

Comment:
I first stumbled over this (thought it should be false) but agree that this is the common behaviour. Not sure how the docs could be improved except with an example.